src/components/Overlay/Overlay.scss

```diff
@@ -0,0 +1,45 @@
+@use '../variables';
+
+$block: '.#{variables.$ns}overlay';
+
+#{$block} {
+    position: absolute;
+    inset: 0;
+
+    display: flex;
+    visibility: hidden;
+    justify-content: center;
+    align-items: center;
+
+    opacity: 0;
+
+    transition:
+        visibility 0.1s linear,
```

Comment:
Seems like it works partially, because without this transition `visibility` changes instantly, but with it it toggles only after 0.1s delay, which is intended. I'll look in other ways to implement it

Reply:
Actually, mozilla's docs state that visibility is animatable, but its interpolation handled as discrete: https://developer.mozilla.org/en-US/docs/Web/CSS/visibility#interpolation